easybuild/framework/easyblock.py

```diff
@@ -1438,6 +1438,27 @@ def make_module_extra(self, altroot=None, altversion=None):
                                      value, type(value))
             lines.append(self.module_generator.append_paths(key, value, allow_abs=self.cfg['allow_append_abs_path']))
 
+        # Add automatic PYTHONPATH or EBPYTHONPREFIXES if they aren't already present and python paths exist
+        if not os.path.isfile(os.path.join(self.installdir, 'bin/python')):  # only needed when not a python install
+            candidate_paths = (os.path.relpath(path, self.installdir)
+                               for path in glob.glob(f'{self.installdir}/lib/python*/site-packages'))
+            python_paths = [path for path in candidate_paths if re.match(r'lib/python\d+\.\d+/site-packages', path)]
+
+            runtime_deps = [dep['name'] for dep in self.cfg.dependencies(runtime_only=True)]
+            use_ebpythonprefixes = 'Python' in runtime_deps and \
+                build_option('prefer_python_search_path') == EBPYTHONPREFIXES and not self.cfg['force_pythonpath']
+
+            if python_paths:
+                # Add paths unless they were already added
+                if use_ebpythonprefixes:
+                    path = ''  # EBPYTHONPREFIXES are relative to the install dir
+                    if path not in self.module_generator.added_paths_per_key[EBPYTHONPREFIXES]:
+                        lines.append(self.module_generator.prepend_paths(EBPYTHONPREFIXES, path))
+                else:
+                    for python_path in python_paths:
```

Comment:
I'm wondering: There can/should never be more than one here, can there? It would mean we'd have e.g. `lib/python3.8` and `lib/python2.6` and will add both to $PYTHONPATH which can't be right

Reply:
A bit of history, went through a few iterations here, at some points i had lib/ and lib64/, which if not symlinked, could be both considered without being an error (and there was a few cases like that which falsely triggered an error), so I just got tired of dealing with that as well so I opted skip that error checking for now.

Since then, I've changed to only consider lib/ since that's the only thing EBPYTHONPREFIXES could handle anyway, so, yes, it's changed slightly. So yes, if someone makes a non PythonPackage/PythonBundle, which also does multi_deps for python, and they set `force_python = True` for some reason, there would be multiple pythonpaths which maybe (possible likely) would break things (though breaking things isn't guaranteed if the actual python modules installed happens to work with both).

Reply:
So when using `$EBPYTHONPREFIX`, it's fine if there are multiple ones (there would be when using `multi_deps`), but not when using `$PYTHONPATH`, so we should make the latter an error.

I'll look into that, I'm working on this right now (implementing tests)